File: <100/048/048.py
import sys


sys.setrecursionlimit(2000)
mod = 10**10


def naive_pow(base: int, power: int) -> int:
    if power == 1: return base % mod
    return base * naive_pow(base, power-1) % mod


def main() -> None:
    answer: int = 0
    for i in range(1, 1001):
        answer = (answer + naive_pow(i, i)) % mod

    print(answer)
    return


if __name__ == '__main__':
    main()

File: <100/048/048.cpp
#include<iostream>
using namespace std;
using ll = int64_t;

const ll mod = 1e10;
ll answer;

ll naive_pow(ll base, ll power){
  if(power == 1) return base % mod;
  return base * naive_pow(base, power-1) % mod;
}

int main(){
  for(ll i = 1; i <= 1000; ++i){
    answer = (answer + naive_pow(i, i)) % mod;
  }

  cout << answer << endl;
}